src/io.rs

```diff
@@ -225,11 +235,28 @@ pub trait Read<'a> {
     fn remaining_data(&self) -> usize;
 
     /// Reads a little-endian integer.
-    fn read_le<I: LeInt>(&mut self) -> Result<I, Error> {
+    fn read_le<I: LeInt>(&mut self) -> Result<I, Error>
+    where
+        Self: Sized,
```

Comment:
Q: Why is this constraint required here and in `Write::write_le`?

Reply:
This is so that the traits are object-safe (i.e., `dyn Read<'a>` and `dyn Write` are well-formed types). Generic functions are disallowed on object-safe traits (unsurprising, because it would require runtime monomorphization). The `Self: Sized` bound removes this requirement.

Mind, because `&mut impl Read: Read`, then `&mut dyn Read` can be passed into any function taking `impl Read`, and, because `&mut dyn Read` (unlike `dyn Read`) is `Sized`, you can call `read_le()` on it. I should probably mark the default impl of `read_le` as inline, too, since it's a trivial function and it would cut down on binary bloat...